src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -8950,9 +8951,72 @@ void MacroAssembler::convert_f2l(Register dst, XMMRegister src) {
   bind(done);
 }
 
+void MacroAssembler::round_float(Register dst, XMMRegister src, Register rtmp, Register rcx) {
+  // Following code is exactly mimicking the functionality of java.lang.Math.round(float) method.
+  Label L_special_case, L_block1, L_exit;
+  movl(rtmp, 0x7f800000);
+  movdl(dst, src);
+  andl(dst, rtmp);
+  sarl(dst, 0x17);
+  movl(rtmp, 0x95);
+  subl(rtmp, dst);
+  movl(rcx, rtmp);
+  movl(dst, 0xffffffe0);
+  testl(rtmp, dst);
+  jccb(Assembler::notEqual, L_special_case);
+  movdl(dst, src);
+  andl(dst, 0x7fffff);
+  orl(dst, 0x800000);
+  movdl(rtmp, src);
+  testl(rtmp, rtmp);
+  jccb(Assembler::greaterEqual, L_block1);
+  negl(dst);
+  bind(L_block1);
+  sarl(dst);
+  addl(dst, 0x1);
+  sarl(dst, 0x1);
+  jmp(L_exit);
+  bind(L_special_case);
+  convert_f2i(dst, src);
+  bind(L_exit);
+}
+
+void MacroAssembler::round_double(Register dst, XMMRegister src, Register rtmp, Register rcx) {
```

Comment:
Is it possible to implement this using the similar mxcsr change? In any case comments will help to review round_double and round_float code.

Reply:
LDMXCSR has multi-cycle latency and it will degrade the performance of scalar operation's fast path.